components/ws-daemon/nsinsider/main.go

```diff
@@ -268,22 +269,25 @@ func main() {
 						Name:     "target-pid",
 						Required: true,
 					},
+					&cli.StringFlag{
+						Name:     "workspace-cidr",
+						Required: true,
+					},
 				},
 				Action: func(c *cli.Context) error {
 					containerIf, vethIf, cethIf := "eth0", "veth0", "eth0"
-					mask := net.IPv4Mask(255, 255, 255, 0)
-					vethIp := net.IPNet{
-						IP:   net.IPv4(10, 0, 5, 1),
-						Mask: mask,
-					}
-					cethIp := net.IPNet{
-						IP:   net.IPv4(10, 0, 5, 2),
-						Mask: mask,
+					networkCIDR := c.String("workspace-cidr")
+
+					vethIp, cethIp, mask, err := processWorkspaceCIDR(networkCIDR)
+					if err != nil {
+						return xerrors.Errorf("parsing workspace CIDR (%v):%v", networkCIDR, err)
 					}
-					masqueradeAddr := net.IPNet{
-						IP:   vethIp.IP.Mask(mask),
-						Mask: mask,
+
+					vethIpNet := net.IPNet{
+						IP:   net.ParseIP(vethIp.String()),
```

Comment:
vethIp is already a net.IP

```suggestion
						IP:  IP:   vethIp,
```

Reply:
done